### PR TITLE
Code fix prefix unused2

### LINKED ISSUE #14420: QuickFix for unused paramter errors should also offer renaming the paramter to start with `_`
```ts
function f (a) {
    console.log(JSON.stringify(arguments));
}
```

Should be turned to:

```ts
function f (_a) {
    console.log(JSON.stringify(arguments));
}
```

### PR DESCRIPTION
**Pull request by aozgaa**
Fixes #14420.

Note that we do not add underscores for parameter declarations or imports, as underscores do no work for suppressing unused errors for these constructs.

### CODE REVIEW

File: src/services/codefixes/fixUnusedIdentifier.ts [hunk 1]
```diff
@@ -18,7 +18,9 @@ namespace ts.codefix {
 
             switch (token.kind) {
                 case ts.SyntaxKind.Identifier:
-                    return deleteIdentifier();
+                    let actions = deleteIdentifier(<Identifier>token);
+                    (actions || (actions = [])).push(prefixIdentifierWithUnderscore(<Identifier>token));
```
Comment (mhegazy) [inline src/services/codefixes/fixUnusedIdentifier.ts]: we should not do this all the time. only for parameters and for..of/for .. in variables

File: tests/cases/fourslash/unusedClassInNamespace1AddUnderscore.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/// <reference path='fourslash.ts' />
+
+// @noUnusedLocals: true
+//// [| namespace greeter {
+////   class class1 {
+////   }
+//// } |]
+
+verify.rangeAfterCodeFix(`namespace greeter {
+    class _class1 { }
```
Comment (mhegazy) [inline tests/cases/fourslash/unusedClassInNamespace1AddUnderscore.ts]: this is still an error.

**Comment by aozgaa** [comment]
@mhegazy can you take another look?

**Comment by mhegazy** [comment]
please port this to release-2.4

**Pull request merged**